orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingPushStrategy.groovy

```diff
@@ -43,17 +43,17 @@ class RollingPushStrategy implements Strategy {
 
   @Override
   List<Stage> composeBeforeStages(Stage parent) {
-    return composeFlow(parent)
-      .findAll({ it.syntheticStageOwner == SyntheticStageOwner.STAGE_BEFORE })
-  }
+    def source = sourceResolver.getSource(parent)
 
-  @Override
-  List<Stage> composeAfterStages(Stage parent) {
-    return composeFlow(parent)
-      .findAll({ it.syntheticStageOwner == SyntheticStageOwner.STAGE_AFTER })
+    if (!source) {
+      throw new IllegalArgumentException("Could not find source server group for rolling push. Does the specified cluster exist?")
```

Comment:
Thoughts on using `SpinnakerException`?

Reply:
yes, thank you!

Reply:
actually, prob a userexception, this would happen if they try to rolling push without a prior asg